### PR TITLE
Flaky test issue automation

### LINKED ISSUE #43594: Flake Issue Creator
Github action that runs nightly and 
- uses metabase API key to check a flake question
- for the top X flakes, queries open issues with the exact title `[Flake Fix] {testName}`
- if no such issue is open, it opens it
- if such an issue is open post a follow-up comment that it's still flaky

Questions
- how to assign to teams?

### PR DESCRIPTION
**Pull request by iethree**
Closes https://github.com/metabase/metabase/issues/43594

### Flake Issue Creator

This script is intended to be run nightly, and pulls data from our metabase instance on flaky tests, and uses that to create github issues to fix those flaky tests, if none already exist for that test. 

Screen Shot 2024-06-07 at 4 09 54 PM

If it finds an issue that remains flaky, it will post update comments with updated flake information.

Screen Shot 2024-06-07 at 4 10 16 PM

### Flake Summary Message

Probably weekly, send a message showing numbers of recentlly fixed flakes and open flake issues to bring visibility to the flake situation.

Screen Shot 2024-06-07 at 3 44 52 PM

### CODE REVIEW

File: .github/workflows/flake-issue-creator.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Flake Issue Creator
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 6 * * 1-5' # every weekday at 1am US eastern time
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply .github/workflows/flake-issue-creator.yml:6]: I think runners' TZ is "US/Pacific"
https://github.com/metabase/metabase/blob/master/.github/workflows/e2e-tests.yml#L132

[the remark this answers is not in the document]
Comment (iethree) [reply .github/workflows/flake-issue-creator.yml:6]: the cron is UTC, so it'll be 1/2am EST depending on DST

Screen Shot 2024-07-10 at 4 19 42 AM

**Comment by tsmacdonald** [comment]
It looks reasonable but I don't feel empowered to give you a real :ballot_box_with_check:

**Review by crisptrutski — APPROVED** [review]
Structure and logic LGTM

**Pull request merged**